pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -129,6 +130,9 @@ public QueryPlan planQuery(String sqlQuery, SqlNodeAndOptions sqlNodeAndOptions)
       plannerContext.setOptions(sqlNodeAndOptions.getOptions());
       RelRoot relRoot = compileQuery(sqlNodeAndOptions.getSqlNode(), plannerContext);
       return toDispatchablePlan(relRoot, plannerContext);
+    } catch (CalciteContextException e) {
+      throw new RuntimeException("Error composing query plan for '" + sqlQuery
+          + "': " + e.getMessage() + "'", e);
```

Comment:
could we add a better message? what's the benefit for adding this extra catch clause?

Reply:
This allows us to bubble up the error from the underlying message. e.g. instead of the user seeing `"Error composing query plan for : 'some sql'"` they'll see more details in the case of a `CalciteContextException`, which will give info like `Cannot use operator '=' to compare <STRING> and <INTEGER>`. I didn't want to bubble up any random exception because some might have really unehlpeful error messages, thats' why I separate out the clauses.

Reply:
hmm. I am not 100% sure i understand. `e.getMessage()` is also included in the `e` part which is also part of the other catch clause branch below correct? 

or the other way to ask is why can't we add e.getMessage() to both branches of catch exception?

Reply:
the problem is that what's shown to the user isn't the entire stack trace, that part gets swallowed upstream somewhere (and that makes sense, that would make for a bad user experience to see the entire stack trace and causedBy). All they see is what's in the error message.

> or the other way to ask is why can't we add e.getMessage() to both branches of catch exception?

that's what I was saying earlier. I don't want to add the cause if the exception makes no sense (like a `NullPointerException`) - the user shouldn't see that kind of message as it really isn't helpful.